spec/models/child_benefit_tax_calculator_spec.rb

```diff
@@ -121,6 +121,25 @@
         ).has_errors?).to eq(false)
       end
     end
+
+    describe "#is_part_year_claim" do
```

Comment:
I find it a bit confusing that for validating the `:is_part_year_claim` we need to setup the attributes: `:children_count` and `:year`

Would something like this work for you:
```ruby
it "should not contain errors if tax claim duration is set to no" do
  calc = ChildBenefitTaxCalculator.new(is_part_year_claim: "no")
  calc.valid?

  expect(calc.errors).to_not have_key(:is_part_year_claim)
end
```